src/allmydata/storage/http_server.py

```diff
@@ -0,0 +1,94 @@
+"""
+HTTP server for storage.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from future.utils import PY2
+
+if PY2:
+    # fmt: off
+    from future.builtins import filter, map, zip, ascii, chr, hex, input, next, oct, open, pow, round, super, bytes, dict, list, object, range, str, max, min  # noqa: F401
+    # fmt: on
+
+from functools import wraps
+
+from klein import Klein
+from twisted.web import http
+
+# TODO Make sure to use pure Python versions?
+from cbor2 import dumps
+
+from .server import StorageServer
+from .http_client import swissnum_auth_header
+
+
+def _authorization_decorator(f):
+    """
+    Check the ``Authorization`` header, and (TODO: in later revision of code)
+    extract ``X-Tahoe-Authorization`` headers and pass them in.
+    """
+
+    @wraps(f)
+    def route(self, request, *args, **kwargs):
+        if request.requestHeaders.getRawHeaders("Authorization", [None])[0] != str(
+            swissnum_auth_header(self._swissnum), "ascii"
+        ):
+            request.setResponseCode(http.UNAUTHORIZED)
+            return b""
+        # authorization = request.requestHeaders.getRawHeaders("X-Tahoe-Authorization", [])
+        # For now, just a placeholder:
+        authorization = None
+        return f(self, request, authorization, *args, **kwargs)
+
+    return route
+
+
+def _route(app, *route_args, **route_kwargs):
```

Comment:
Maybe `_authorized_route` or smilar would make it more-clear when reading code?

Reply:
Makes sense, yeah.